src/validation.cpp

```diff
@@ -347,7 +347,7 @@ static bool IsCurrentForFeeEstimation() EXCLUSIVE_LOCKS_REQUIRED(cs_main)
     AssertLockHeld(cs_main);
     if (::ChainstateActive().IsInitialBlockDownload())
         return false;
-    if (::ChainActive().Tip()->GetBlockTime() < (GetTime() - MAX_FEE_ESTIMATION_TIP_AGE))
+    if (::ChainActive().Tip()->GetBlockTime() < (GetTime<std::chrono::seconds>() - MAX_FEE_ESTIMATION_TIP_AGE).count())
```

Comment:
```suggestion
    if (::ChainActive().Tip()->GetBlockTime() < count_seconds(GetTime<std::chrono::seconds>() - MAX_FEE_ESTIMATION_TIP_AGE))
```
Otherwise `.count` will return the wrong value in a future refactor where the second operand of the `-` changes its type to something with greater precision than seconds.

Reply:
Done. It'd be great if you could open an issue documenting what these future, higher-precision time refactor changes are going to be, so it's clearer why we are writing code to accommodate them now.

Reply:
There are no specific refactors planned. The mentioned refactor was imaginary to illustrate why writing code as robust as possible is important.